### PR TITLE
Upgrade Redis to v7 and upgrade BullMQ to latest

### PR DESCRIPTION
**Pull request by dotansimha**
Closes https://github.com/kamilkisiela/graphql-hive/issues/901  https://github.com/kamilkisiela/graphql-hive/pull/920 https://github.com/kamilkisiela/graphql-hive/pull/800

## TODO

- [x] Upgrade Redis image in both `community` and `dev` Compose files 
- [x] Upgrade Redis image in deployment (and use it from `.community` file) 
- [x] Upgrade to BullMQ v3 and fix BullMQ patch
- [x] Refactor services that uses `QueueScheduler` (no longer needed) 
- [x] Remove unused `BULLMQ_COMMANDS_FROM_ROOT` 
- [x] Drop `hostPort` from Redis deployment to K8s to avoid downtime when switching Redis versions 

## Testing
- [x] `dev` round 1 (https://github.com/the-guild-org/graphql-hive-deployment/actions/runs/3821723200/jobs/6501150864)
- [x] `dev` round 2 without `hostPort` (https://github.com/the-guild-org/graphql-hive-deployment/actions/runs/3821942402)
- [x] `staging` (https://github.com/the-guild-org/graphql-hive-deployment/actions/runs/3822017798)

### CODE REVIEW

File: patches/bullmq@3.5.1.patch [hunk 1]
```diff
@@ -1,18 +1,5 @@
-diff --git a/dist/esm/classes/redis-connection.js b/dist/esm/classes/redis-connection.js
-index 7843e65..3807efe 100644
---- a/dist/esm/classes/redis-connection.js
-+++ b/dist/esm/classes/redis-connection.js
-@@ -105,7 +105,7 @@ export class RedisConnection extends EventEmitter {
-     }
-     loadCommands() {
-         return (this._client['bullmq:loadingCommands'] ||
--            (this._client['bullmq:loadingCommands'] = scriptLoader.load(this._client, path.join(__dirname, '../commands'))));
-+            (this._client['bullmq:loadingCommands'] = scriptLoader.load(this._client, path.join(__dirname, process.env.BULLMQ_COMMANDS_FROM_ROOT ? './' : '../commands'))));
```
Comment (kamilkisiela) [inline patches/bullmq@3.5.1.patch:10]: Did you test it on staging?

Comment (dotansimha) [reply patches/bullmq@3.5.1.patch:10]: I think v3 is no longer uses the `commands` in that form, but let me double check.

**Review by kamilkisiela — APPROVED** [review]

File: patches/bullmq@3.5.1.patch (hunk 1 above)
Comment (kamilkisiela) [reply patches/bullmq@3.5.1.patch:10]: If it works in integration tests, it will work on the cluster, but just wanted to make sure :)

Comment (dotansimha) [reply patches/bullmq@3.5.1.patch:10]: tested, all good ✅

**Pull request merged**